src/OpenTelemetry.Extensions.AzureMonitor/ApplicationInsightsSampler.cs

```diff
@@ -71,7 +71,7 @@ public override SamplingResult ShouldSample(in SamplingParameters samplingParame
             return this.recordAndSampleSamplingResult;
         }
 
-        double sampleScore = DJB2SampleScore(samplingParameters.TraceId.ToHexString().ToLowerInvariant());
+        double sampleScore = DJB2SampleScore(samplingParameters.TraceId.ToHexString().ToUpperInvariant());
```

Comment:
This showed up on code analysis: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1308